### PR TITLE
feat: add sign and broadcast hook

### PR DESCRIPTION
**Pull request by joshuanatanielnm**
This is draft to add sign and broadcast hook, from [here](https://cosmos.github.io/cosmjs/latest/stargate/classes/SigningStargateClient.html#signAndBroadcast)

What left is the docs for this hook

### CODE REVIEW

File: packages/graz/src/actions/methods.ts [hunk 1]
```diff
@@ -104,6 +104,24 @@ export const sendIbcTokens = async ({
...
 
+// https://cosmos.github.io/cosmjs/latest/stargate/classes/SigningStargateClient.html#signAndBroadcast
+export interface SignAndBroadcastArgs {
+  signerAddress?: string;
+  messages: readonly EncodeObject[];
+  fee: number | StdFee | "auto";
+  memo?: string;
+}
+
+export const SignAndBroadcast = async ({ signerAddress, messages, fee, memo }: SignAndBroadcastArgs) => {
```
Comment (codingki) [inline packages/graz/src/actions/methods.ts:116]: this is a function, please use lowercase on the first letter `signAndBroadcast`

Comment (codingki) [reply packages/graz/src/actions/methods.ts:116]: I don't think we don't need this function, we can execute it inside the mutation

File: packages/graz/src/hooks/methods.ts [hunk 2]
```diff
@@ -115,6 +118,30 @@ export const useSendIbcTokens = ({
     status: mutation.status,
   };
 };
+export const useSignAndBroadcast = async ({
+  onError,
+  onLoading,
+  onSuccess,
+}: MutationEventArgs<SignAndBroadcastArgs, DeliverTxResponse> = {}) => {
+  const { signer } = useOfflineSigners();
+  const signerAddress = signer ? (await signer.getAccounts())[0]?.address : "";
```
Comment (codingki) [inline packages/graz/src/hooks/methods.ts:127]: if the `signerAddress` is undefined throw an error

**Review by codingki — CHANGES_REQUESTED** [review]

File: packages/graz/src/actions/methods.ts (hunk 1 above)
Comment (joshuanatanielnm) [reply packages/graz/src/actions/methods.ts:116]: just want to confirm, what you mean is we don't need `signAndBroadcast ` function?

Comment (codingki) [reply packages/graz/src/actions/methods.ts:116]: yes, please check `sendTokens` as reference

**Pull request closed without merging**